resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -307,6 +301,9 @@ private[spark] abstract class YarnSchedulerBackend(
 
       case RetrieveLastAllocatedExecutorId =>
         context.reply(currentExecutorIdCounter)
+
+      case RetrieveLastRequestExecutors =>
```

Comment:
Why make a new call for this? What I had in mind would re-use `RetrieveLastAllocatedExecutorId` adding a new properties that tell the AM what's the initial state. Might be good to change the name of the message at that point to (e.g. `GetAMInitialState` or something).

That call is already made during initialization of `YarnAllocator`, so it should be a smaller change overall.

So, you'd have one already existing RPC instead of one RPC that may cause the AM to send another RPC to signal the driver to send an RPC to the AM (my head is spinning now).